packages/nimble-components/src/table/tests/table.mdx

```diff
@@ -16,6 +16,42 @@ import * as tableStories from './table.stories';
 <DocsStory of={tableStories.table} expanded={false} />
 <Controls of={tableStories.table} />
 
+## Delay-loaded Hierarchy
+
+<Description of={tableStories.delayedHierarchy} />
+<DocsStory of={tableStories.delayedHierarchy} expanded={false} />
+<Controls of={tableStories.delayedHierarchy} />
+
+### Delay-loaded Hierarchy Usage
+
+The expected usage of the dynamically loaded hierarchy is as follows:
+
+1. Configure `idFieldName` and `parentIdFieldName` on the table
```

Comment:
Note: This expected usage is copied directly from the HLD. I decided to leave in the documentation that included `loadingChildren` to avoid extra churn here and since I already added a note saying that the loading state wasn't implemented yet, but I could also see the argument that we should adjust these steps based on what is implemented today.